### PR TITLE
Fix image durations

### PR DESCRIPTION
**Pull request by NoSloppy**
Then these could be put in the default config.ini document:

```
# Below are the display durations in milliseconds.
# If commented out or not specified, the default is used.
# As of ProffieOS 7, effect images and/or font "name" text
# can use 0.0 to automatically sync to the effect sound file duration.

# Boot image duration
# Uses the specified duration,
# falls back to the Font image duration if not specified,
# or defaults to 3500ms.
ProffieOSBootImageDuration=4000.0

# Font image duration
# Default 3000.0
ProffieOSFontImageDuration=0.0

# Text Message duration 
# Used primarily for preset "name", but also Startup screen text and errors.
# Uses the specified duration,
# falls back to the Font image duration if not specified,
# or defaults to 3000ms.
# *Note* Startup text will display for a minimum of 3500ms to ensure readability.
ProffieOSTextMessageDuration=3500.0
```

### CODE REVIEW

File: display/ssd1306.h [hunk 1]
```diff
@@ -395,15 +404,19 @@ class StandardDisplayController : public DisplayControllerBase<Width, col_t>, Sa
...
       case SCREEN_ERROR_MESSAGE:
       case SCREEN_MESSAGE: {
-	uint32_t t;
+        uint32_t text_time;
         if (font_config.ProffieOSTextMessageDuration != -1) {
-	  t = font_config.ProffieOSTextMessageDuration;
-        } else if (font_config.ProffieOSFontImageDuration > 0) {
-	  t = font_config.ProffieOSFontImageDuration;
+          text_time = (font_config.ProffieOSTextMessageDuration == 0 ?
+                       SaberBase::sound_length * 1000 :
```
Comment (profezzorn) [inline display/ssd1306.h:410]: We can't use SaberBase::sound_length here.
It's going to have basically random values outside of SB_*2.

Also, I'm not sure that we even want to do that SCREEN_ERROR_MESSAGE.

**Review by profezzorn — COMMENTED** [review]
I'm not entirely sure what the problem is that is being fixed here.
It seems like a fair number of changes, and it's not easy to understand the motivation behind them.

**Comment by NoSloppy** [comment]
> I'm not entirely sure what the problem is that is being fixed here. 

I can answer. The problem at first was that on.bmp was preventing boot, font, and the text from showing.
Last PR was a fix, but it wasn't entirely right for all scenarios.
Apparently having no specified textMessageDuration caused it not to show.
The defaut `-1` value was being compared to `t_`, the time since we last switched screens, so essentially it was ignored and on.bmp would kick in immediately.
So while tracking that down, I noticed "what if" the user also had no font or boot durations set, then considered the wav sync and it kind of snowballed into adding in some checks to make it work the way I understood it should work. I can answer the other questions below in place.

File: display/ssd1306.h [hunk 2]
```diff
@@ -366,10 +366,19 @@ class StandardDisplayController : public DisplayControllerBase<Width, col_t>, Sa
           display_->DrawText(install_time,0,63, Starjedi10pt7bGlyphs);
         }
         next_screen_ = SCREEN_PLI;
-        if (font_config.ProffieOSTextMessageDuration != -1) {
-          return font_config.ProffieOSTextMessageDuration;
-        } else if (font_config.ProffieOSFontImageDuration > 0) {
-          return font_config.ProffieOSFontImageDuration;
+        // Give user option to set longer than default startup screen duration.
+        // but ensure startup info has enough time to be read.
+        // Prefer TextMessageDuration. If set < 3500 default, override with default value.
```
[the remark this answers is not in the document]
Comment (NoSloppy) [reply display/ssd1306.h:371]: I figured the startup screen is "important" enough to make sure it has adequate screen time, so a minimum duration would provide that.

**Comment by profezzorn** [comment]
The font image duration has a default value, so the user would have to explicitly set it to something stupid.
We may want to consider generating an error of some sort if the user provides a stupid value, but we should not "fix" it, as that will just hide the problem and confuse users mode.

File: display/ssd1306.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (profezzorn) [reply display/ssd1306.h:371]: No, leave it alone.

**Comment by NoSloppy** [comment]
They all have defaults. All I did really was primarily make the on.bmp "delay" work because the durations are right.
Then, added a check in case the duration was set to zero, where the user is syncing to the sound, and moved the default fontImageDuration from hybrid_font into the code like the other 2 so the GetCorrectDuration function could work.

The comments should be clear about what's happening. I know it's somewhat ternary hell, and if there's a better way I wouldn't be surprised. This is working, but I know, the question is does it make sense. I think it does.

File: display/ssd1306.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (NoSloppy) [reply display/ssd1306.h:371]: Oh - Sorry I see what you mean now.
it already defaults to 3500 if the TextMessageDuration is set to 0 for sync. And if it's set to something stupid like 3, that's just silly on the user. Ok cool.

File: display/ssd1306.h (hunk 1 above)
Comment (NoSloppy) [reply display/ssd1306.h:401]: oh i see. error message case falls through. hmm.
Would a new function based on ShowFileWithSoundLength() be appropriate for text like ShowTextWithSoundLength?
it's really only the preset "name" argument that could use it. 

Or not, and just set a hard time and don't sync to sound i guess. :(

Comment (profezzorn) [reply display/ssd1306.h:401]: It seems reasonable.
Is this stuff related to the SB_On fix though?
If the answer is "not really", maybe split it into two PRs?

Comment (NoSloppy) [reply display/ssd1306.h:401]: Sure totally. The goal here was to rectify where the added code in SB_On() did not work if text duration wasn't set. It's really the comparison to `t_` that's not correct.

File: display/ssd1306.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (NoSloppy) [reply display/ssd1306.h]: reverted

**Comment by NoSloppy** [comment]
closing and resubmitting cleaner

**Pull request closed without merging**